ui/app/contexts/metametrics.new.js

```diff
@@ -30,12 +32,39 @@ export const MetaMetricsContext = createContext(() => {
 
 const PATHS_TO_CHECK = Object.keys(PATH_NAME_MAP)
 
+function useSegmentContext () {
+  const match = useRouteMatch({ path: PATHS_TO_CHECK, exact: true, strict: true })
+  const locale = useSelector(getCurrentLocale)
+  const txData = useSelector(txDataSelector) || {}
+  const confirmTransactionOrigin = txData.origin
+
+  const referrer = confirmTransactionOrigin ? {
+    url: confirmTransactionOrigin,
+  } : undefined
+
+  return {
+    app: {
+      version: `${global.platform.getVersion()}-${process.env.METAMASK_ENVIRONMENT}`,
+      name: 'MetaMask Extension',
+    },
+    locale,
```

Comment:
We use underscores instead of dashes for the locale code, which is not compliant with the language tag spec. It looks like Segment also expects dashes instead of underscores ([see the `locale` field in the context docs](https://segment.com/docs/connections/spec/common/#context)). So we should do a substitution here before we send this in.